mmv1/third_party/terraform/services/sql/resource_sql_database_instance_test.go

```diff
@@ -2361,6 +2364,71 @@ func TestAccSqlDatabaseInstance_ReplicaPromoteSkippedWithNoMasterInstanceNameAnd
 	})
 }
 
+// Switchover between primary and cascadable replica sunny case
+func TestAccSqlDatabaseInstance_SwitchoverSuccess(t *testing.T) {
+	t.Parallel()
+	primaryName := "tf-test-sql-instance-" + acctest.RandString(t, 10)
+	replicaName := "tf-test-sql-instance-replica" + acctest.RandString(t, 10)
+	acctest.VcrTest(t, resource.TestCase{
+		PreCheck:                 func() { acctest.AccTestPreCheck(t) },
+		ProtoV5ProviderFactories: acctest.ProtoV5ProviderFactories(t),
+		CheckDestroy:             testAccSqlDatabaseInstanceDestroyProducer(t),
+		Steps: []resource.TestStep{
+			{
+				Config: testGoogleSqlDatabaseInstanceConfig_SqlServerwithCascadableReplica(primaryName, replicaName),
+				// TODO: remove when API change is rolled out.
+				ExpectNonEmptyPlan: true,
+			},
+			{
+				ResourceName:            "google_sql_database_instance.original-primary",
+				ImportState:             true,
+				ImportStateVerify:       true,
+				ImportStateVerifyIgnore: ignoredReplicaConfigurationFields,
+			},
+			{
+				ResourceName:            "google_sql_database_instance.original-replica",
+				ImportState:             true,
+				ImportStateVerify:       true,
+				ImportStateVerifyIgnore: ignoredReplicaConfigurationFields,
+			},
+			{
+				Config: googleSqlDatabaseInstance_switchover(primaryName, replicaName),
```

Comment:
Oh, on second thought because this is sql_database_instance you don't need to do that because we can't create a DB with the same name immediately after deleting one. So this shouldn't be a problem as long as we don't see name conflict errors coming from the test